_tools/generate_release_cycle.py

```diff
@@ -28,21 +28,52 @@ def parse_date(date_str: str) -> dt.date:
 class Versions:
     """For converting JSON to CSV and SVG."""
 
-    def __init__(self) -> None:
+    def __init__(self, limit_to_active=False, special_py27=False) -> None:
         with open("include/release-cycle.json", encoding="UTF-8") as in_file:
             self.versions = json.load(in_file)
 
         # Generate a few additional fields
         for key, version in self.versions.items():
             version["key"] = key
-            version["first_release_date"] = parse_date(version["first_release"])
+            version["first_release_date"] = r1 = parse_date(version["first_release"])
+            version["start_security_date"] = r1 + dt.timedelta(days=2 * 365)
```

Comment:
It's only two years for 3.13+:

> Note: 2 years of full support start with Python 3.13. Python versions 3.9 - 3.12 operate on a calendar with 1½ year of full support, followed by 3½ more years of security fixes.

https://peps.python.org/pep-0602/#years-of-full-support-3-more-years-of-security-fixes

Reply:
Should we add an actual security date to the JSON?  The PEPs don't have dates for this transition.

Reply:
Another option: hardcode the 1.5-year vs 2-year delta based on the version number.

Reply:
I chose the hard-coded option.

Reply:
Hard-coding is good for now.

I'm thinking about having all the release dates in a reusable JSON file, but that's for the future:

https://github.com/python/pythondotorg/issues/2701#issuecomment-2730267006